src/main/groovy/com/palantir/gradle/docker/DockerExtension.groovy

```diff
@@ -118,8 +119,13 @@ class DockerExtension {
         Preconditions.checkArgument(!Strings.isNullOrEmpty(name),
             "name is a required docker configuration item.")
 
-        resolvedDockerfile = project.file(dockerfile)
-        Preconditions.checkArgument(resolvedDockerfile.exists(), "dockerfile '%s' does not exist.", dockerfile)
+        if (dockerfile != null) {
+            resolvedDockerfile = dockerfile
+        } else {
+            resolvedDockerfile = project.file(dockerfilePath)
+            Preconditions.checkArgument(resolvedDockerfile.exists(),
```

Comment:
I agree, but I think moving validation of the paths to task execution is enough work that it isn't worth dealing with it in this PR.  The docker daemon will throw a useful enough error here to let someone with a misconfigured plugin know what has gone wrong, which the purpose of this error.